core/server/api/canary/members.js

```diff
@@ -26,8 +26,50 @@ const cleanupUndefined = (obj) => {
     }
 };
 
-// NOTE: this method can be removed once unique constraints are introduced ref.: https://github.com/TryGhost/Ghost/blob/e277c6b/core/server/data/schema/schema.js#L339
-const sanitizeInput = (members) => {
+const sanitizeInput = async (members) => {
+    const validationErrors = [];
+    let invalidCount = 0;
+
+    const jsonSchema = require('./utils/validators/utils/json-schema');
+    const schema = require('./utils/validators/input/schemas/members-upload');
+    const definitions = require('./utils/validators/input/schemas/members');
+
+    let invalidValidationCount = 0;
+    try {
+        await jsonSchema.validate(schema, definitions, members);
+    } catch (error) {
+        if (error.errorDetails && error.errorDetails.length) {
+            const jsonPointerIndexRegex = /\[(?<index>\d+)\]/;
+
+            let invalidRecordIndexes = error.errorDetails.map((errorDetail) => {
+                if (errorDetail.dataPath) {
+                    const key = errorDetail.dataPath.split('.').pop();
+                    const [, index] = errorDetail.dataPath.match(jsonPointerIndexRegex);
+                    validationErrors.push(new errors.ValidationError({
+                        message: i18n.t('notices.data.validation.index.schemaValidationFailed', {
+                            key
+                        }),
+                        context: `${key} ${errorDetail.message}`,
+                        errorDetails: `${errorDetail.dataPath} with value ${members[index][key]}`
+                    }));
+
+                    return Number(index);
+                }
+            });
+
+            invalidRecordIndexes = _.uniq(invalidRecordIndexes);
+            invalidRecordIndexes = invalidRecordIndexes.filter(index => (index !== undefined));
+
+            invalidRecordIndexes.forEach((index) => {
```

Comment:
This looks hacky. But here's why I went with this approach. After schema validation the best we have is indexes of records which fail validations. Didn't find a nicer method to "removeAtIndexes(arr, [...indexes]). Maybe I've missed some nicer way to achieve the same :thinking: